main.go

```diff
@@ -77,17 +77,39 @@ func main() {
 			queryString = fmt.Sprintf("?%s", strings.TrimLeft(req.QueryString, "?"))
 		}
 
-		functionURL := fmt.Sprintf("http://%s%s:8080%s%s", req.Function, config.FunctionSuffix, path, queryString)
+		functionURL := fmt.Sprintf("http://%s%s:8080%s%s",
+			req.Function,
+			config.FunctionSuffix,
+			path,
+			queryString)
+
+		if config.GatewayInvoke {
+			functionURL = fmt.Sprintf("http://%s:8080/function/%s/%s%s",
+				config.GatewayAddress,
+				req.Function,
+				path,
+				queryString)
+		}
 
+		start := time.Now()
 		request, err := http.NewRequest(http.MethodPost, functionURL, bytes.NewReader(req.Body))
```

Comment:
@alexellis Quick question. Should'n the method be `req.Method` instead of `http.MethodPost`?

Reply:
Asynchronous invocations are only POST at this time. This also didn't change in the patch.